extension/js/common/helpers.ts

```diff
@@ -11,16 +11,24 @@ import { ContactStore } from './platform/store/contact-store.js';
 import { KeyStore } from './platform/store/key-store.js';
 import { PassphraseStore } from './platform/store/passphrase-store.js';
 import { Bm } from './browser/browser-msg.js';
+import { PgpPwd } from './core/crypto/pgp/pgp-password.js';
 export const isFesUsed = async (acctEmail: string) => {
   const { fesUrl } = await AcctStore.get(acctEmail, ['fesUrl']);
   return Boolean(fesUrl);
 };
 
-export const saveKeysAndPassPhrase = async (acctEmail: string, prvs: Key[], ppOptions?: PassphraseOptions) => {
+export const saveKeysAndPassPhrase = async (acctEmail: string, prvs: Key[], ppOptions?: PassphraseOptions, replaceKeys: boolean = false) => {
   const clientConfiguration = await ClientConfiguration.newInstance(acctEmail);
+  if (replaceKeys) {
+    await KeyStore.set(acctEmail, await Promise.all(prvs.map(KeyUtil.keyInfoObj))); // todo: duplicate identities
+    // todo: should we delete passphrases matching the deleted keys?
+  }
   for (const prv of prvs) {
-    await KeyStore.add(acctEmail, prv);
+    if (!replaceKeys) {
+      await KeyStore.add(acctEmail, prv);
+    }
     if (ppOptions !== undefined) {
+      // todo: perhaps it's easier just to store a set of passphrases without specifying longids?
```

Comment:
In situations with many different keys with different pass phrases (manually imported) we'd have to loop-try through pass phrases, which is computationally expensive. Say 3 keys, 3 pass phrases, could be close to 10 attempts (times amount of subkeys to decrypt per key, in a way). So it makes sense to track the associated longid of each pass phrase.